collect_app/src/main/java/org/odk/collect/android/widgets/LikertWidget.java

```diff
@@ -0,0 +1,351 @@
+import org.odk.collect.android.utilities.ViewIds;
+
+import timber.log.Timber;
+
+@SuppressLint("ViewConstructor")
+public class LikertWidget extends ItemsWidget {
+
+    private LinearLayout view;
+    private RadioButton checkedButton;
+    private final LinearLayout.LayoutParams linearLayoutParams = new LinearLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT, 1);
+    private final LayoutParams textViewParams = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
+    private final LayoutParams imageViewParams = new LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
+    private final LayoutParams radioButtonsParams = new LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
+    private final LayoutParams buttonViewParams = new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
+    private final LayoutParams leftLineViewParams = new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 2);
+    private final LayoutParams rightLineViewParams = new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 2);
+
+    HashMap<RadioButton, String> buttonsToName;
+
+    public LikertWidget(Context context, QuestionDetails questionDetails) {
+        super(context, questionDetails);
+
+        setMainViewLayoutParameters();
+        setStructures();
+
+        setButtonListener();
+        setSavedButton();
+        addAnswerView(view);
+    }
+
+    public void setMainViewLayoutParameters() {
+        view = new LinearLayout(getContext());
+        view.setOrientation(LinearLayout.HORIZONTAL);
+        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+        view.setLayoutParams(params);
+    }
+
+    // Inserts the selected button from a saved state
+    public void setSavedButton() {
+        if (getFormEntryPrompt().getAnswerValue() != null) {
+            String name = ((Selection) getFormEntryPrompt().getAnswerValue()
+                    .getValue()).getValue();
+            for (RadioButton bu: buttonsToName.keySet()) {
+                if (buttonsToName.get(bu).equals(name)) {
+                    checkedButton = bu;
+                    checkedButton.setChecked(true);
+                }
+            }
+        }
+    }
+
+    @Override
+    public IAnswerData getAnswer() {
+        if (checkedButton == null) {
+            return null;
+        } else {
+            int selectedIndex = -1;
+            for (int i = 0; i < items.size(); i++) {
+                if (items.get(i).getValue().equals(buttonsToName.get(checkedButton))) {
+                    selectedIndex = i;
+                }
+            }
+            if (selectedIndex == -1) {
+                return null;
+            }
+            SelectChoice sc = items.get(selectedIndex);
+            return new SelectOneData(new Selection(sc));
+        }
+    }
+
+    /**
```

Comment:
This comment comes from the `QuestionWidget` implementation and can be removed or replaced with a comment that describes this specific implementation. Please add `@Override` annotation.

Reply:
I've removed the redundant comment and added the annotation. Don't think the implementation needs a special comment, as the only difference is the [margin](https://github.com/opendatakit/collect/blob/db40fc948a6ea377cefbad3a96217905a3578941/collect_app/src/main/java/org/odk/collect/android/widgets/LikertWidget.java#L120). @andrewsiew2 if you feel there is an appropriate comment, please add it ☺️

Reply:
Agreed it doesn't seem necessary!